METRON-1712: PCAP UI Input validation

tiborm opened the pull request:
## Contributor Comments
This PR contains the input validation for ipSrcAdr, ipSrcPort, ipDestAdr, ipDestPort fiels.

Please find the details in the https://issues.apache.org/jira/browse/METRON-1724 ticket.

There is a followup ticket on this:
Date rage validation https://issues.apache.org/jira/browse/METRON-1724
We'll open a separated PR when it's done.


## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  


In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel).
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?


### For code changes:
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```

- [x] Have you written or updated unit tests and or integration tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

justinleet commented:
Can you please change the title of this PR to "Metron-1712: PCAP UI - Input validation"?

justinleet commented:
I ran this up and I'm able to run a query with ip:
- 999.999.999.999

Each of the four octets should be in the range [0-255], e.g. 0.0.0.0, 255.255.255.255, 255.0.12.100 etc.

I'm also able to run with ports:
- 9000
- 90000
- 900000.9989

Ports can be specified with scientific notation (and apparently send properly!)
- 1e3

Ports with pieces of scientific notation and fairly unreasonable values run with empty strings:
- 1e9
- 1+9
- 1.0e-5

justinleet commented:
Hmm, the port errors seem to have been a browser caching issue on the page.  I looked at again and it validate ports properly.  Sorry for the false alarm.

tiborm commented:
Thanks, Justin! I replaced the regexp for IP validation and added new tests to cover that functionality. Nice catch!

I was unable to reproduce the port validation issue you reported. However, I replaced the regexp and double checked the tests. They seem working fine.

justinleet commented on metron-interface/metron-alerts/src/app/pcap/pcap-filters/pcap-filters.component.spec.ts:
```diff
@@ -331,13 +337,60 @@ describe('PcapFiltersComponent', () => {
+    });
+
+    it('should keep the form enabled if the ip source field is valid', () => {
+      const invalidValues = [
```
Looks like this variable is misnamed, can you rename this (and it's usage in at 382)to `validValues`?

tiborm replied:
Fixed. Thanks!

tiborm commented:
You're right. Copy paste issue. Fixed. Thanks!

justinleet commented:
+1 pending Travis, thanks for the fixes, this is a great quality of life improvement!

merrimanr commented:
This has been merged.  Can you close @tiborm?

tiborm commented:
Thanks for the review!
